### PR TITLE
[Tests] Improve server coverage for Various Services

### PR DESCRIPTION
**Pull request by ivanchimeno**
### Checklist
- [x] I tested *all* changes and *all* related features with different users (student, tutor, editor, instructor, admin) on the test server https://artemistest.ase.in.tum.de.
- [x] Server: I followed the [coding and design guidelines](https://docs.artemis.ase.in.tum.de/dev/guidelines/server/).
- [x] Server: I added multiple integration tests (Spring) related to the features (with a high test coverage)

### Description

This PR improves the code coverage of various services to >= 90%.

### Test Coverage

CourseExamExportService.java 93%
ResultService.java 94%
RepositoryService.java 93%
AuthorizationCheckService.java 90%

### CODE REVIEW

File: src/main/java/de/tum/in/www1/artemis/service/ResultService.java [hunk 1]
```diff
@@ -64,17 +62,6 @@ public ResultService(UserRepository userRepository, ResultRepository resultRepos
...
-     * Sets the assessor field of the given result with the current user and stores these changes to the database. The User object set as assessor gets Groups and Authorities
-     * eagerly loaded.
-     *
-     * @param result Result for which current user is set as an assessor
-     */
-    public void setAssessor(Result result) {
-        User currentUser = userRepository.getUser();
-        result.setAssessor(currentUser);
-    }
-
```
Comment (SimonEntholzer) [inline src/main/java/de/tum/in/www1/artemis/service/ResultService.java:77]: Why did you remove this code? Was it unused? Same question for the block below in the same file.

Comment (ivanchimeno) [reply src/main/java/de/tum/in/www1/artemis/service/ResultService.java:77]: Yes it's not used

File: src/test/java/de/tum/in/www1/artemis/TestRepositoryResourceIntegrationTest.java [hunk 2]
```diff
@@ -338,4 +472,24 @@ public void testGetStatus() throws Exception {
         var receivedStatusAfterCommit = request.get(testRepoBaseUrl + programmingExercise.getId(), HttpStatus.OK, RepositoryStatusDTO.class);
         assertThat(receivedStatusAfterCommit.repositoryStatus.toString()).isEqualTo("CLEAN");
     }
+
+    @Test
+    @WithMockUser(username = "instructor1", roles = "INSTRUCTOR")
+    public void testIsClean() throws Exception {
+        programmingExerciseRepository.save(programmingExercise);
+        doReturn(true).when(gitService).isRepositoryCached(any());
+        var status = request.get(testRepoBaseUrl + programmingExercise.getId(), HttpStatus.OK, LinkedHashMap.class);
```
Comment (FrankeLukas) [inline src/test/java/de/tum/in/www1/artemis/TestRepositoryResourceIntegrationTest.java:481]: Codacy suggests to use the interface instead of `LinkedHashMap`. Is that possible or do we need to use `LinkedHashMap` here?

File: src/test/java/de/tum/in/www1/artemis/util/CourseTestService.java [hunk 3]
```diff
@@ -1326,6 +1328,59 @@ else if (exercise instanceof TextExercise) {
...
+        mockedFiles.when(() -> Files.createDirectories(argThat(path -> path.toString().contains("exports")))).thenThrow(IOException.class);
+        Optional<Path> exportedCourse = courseExamExportService.exportExam(exam.get(), courseArchivesDirPath, exportErrors);
+        mockedFiles.close();
+
+        assertThat(exportedCourse).isEmpty();
+    }
+
+    public void testExportCourseExam_cannotCreateExamsDir() throws Exception {
+        Course course = database.createCourseWithExamAndExercises();
+        List<String> exportErrors = new ArrayList<>();
```
Comment (FrankeLukas) [inline src/test/java/de/tum/in/www1/artemis/util/CourseTestService.java:1372]: Wouldn't it make sense to also check the contents or count of this list in this test case and the ones above?

**Review by FrankeLukas — APPROVED** [review]

File: src/test/java/de/tum/in/www1/artemis/TestRepositoryResourceIntegrationTest.java (hunk 2 above)
Comment (ivanchimeno) [reply src/test/java/de/tum/in/www1/artemis/TestRepositoryResourceIntegrationTest.java]: Unfortunate, I don't know how to resolve this issue. Do you by any chance know how to do it?

Comment (FrankeLukas) [reply src/test/java/de/tum/in/www1/artemis/TestRepositoryResourceIntegrationTest.java]: I think codacy wants us to use HashMap.class here. If that doesn't work we should ignore the warning.

**Comment by imstreetspirit** [comment]
overall seems good except the codacy issue mentioned above

File: src/test/java/de/tum/in/www1/artemis/TestRepositoryResourceIntegrationTest.java (hunk 2 above)
Comment (ivanchimeno) [reply src/test/java/de/tum/in/www1/artemis/TestRepositoryResourceIntegrationTest.java]: Okay I changed it, I hope it removes the warning

**Review by FuchsDominik — APPROVED** [review]
Codacy warning of LinkedHashMap not resolved by using HashMap. Don't know how to resolve this either so would ignore it. Code looks fine and Tests pass 👍

**Review by FrankeLukas — APPROVED** [review]
Lgtm. We could try using the class HashMap inherits from (I think some sort of AbstractMap?) but that would be more confusing when reading the code imo.
